### ISSUE TITLE
Suggestions

### ISSUE THREAD

**Issue by ocjojo** [bluemoehre/chrome-meta-debugger#4]
First of all thanks for this extension.

What are your thoughts on providing meta tag suggestions?
I would quite like to have suggestions for common SEO metatags, like favicon, description or open graph tags.
Maybe these suggestions could be placed at the end of the list or shown using a toggle.

**Comment by bluemoehre** [comment]
I am happy to hear you find it useful =)

Well, it was never planned to add suggestions for SEO metatags since SEO is a very dynamic thing and this tool should just be an inspector. I am afraid I won't have the time to updated these suggestions regularly to reflect current best practices.

But it could be useful to notify the user if some really basic things like description / title / favicon are missing.
These could be easily implemented and will last a long time.

Thanks for your feedback!

**Comment by frouo** [comment]
Would love that too but I understand it's a lot of work and kind of another tool.

I have made this quick hack to suit my needs. I won't make a PR since it's far from being a scalable solution. Here is the commit if it can help someone : https://github.com/frouo/chrome-meta-debugger/commit/082adefa35de63d08daaaa077e73e9d62fc4a1cb

<img width="276" alt="image" src="https://user-images.githubusercontent.com/2499356/160259668-f1b16900-d293-47ba-b9f2-d48517b62796.png">

**Comment by bluemoehre** [comment]
Would it help to simply show the string length of these both tags? This would be kinda generic.

**Comment by frouo** [comment]
Not really, at least for me. What are the most important (for me) are 👍 /❌ where I can get in a fraction of a second if the tag is ok. Note that I did not use ✅ emoji because the green is too close to the [META] label and it took my brain few more millisecs to process 😅

In case it's a ❌, I can see how far or close I am from the target depending on the length.

I was thinking about this solution: a place where (extension settings?) where one could write a piece of JS logic. So one could enrich the displayed data as wanted.

For example to achieve the same result of the screenshot I wish I could have write somewhere:

```
(tag, value) => {
}
```

where `tag` would be either `title`, `description` etc. and value the value of the metatag. What do you think?

**Comment by bluemoehre** [comment]
Yeh, using some colors/symbols for a fast scan is indeed a better solution rather than text only.

I don't know yet if it's unproblematic to "_inject_" code from the user in terms of restrictions/security policies etc.. 
Allow hooking in the render function of a row would be a very generic option for full flexibility, but also is really advanced and requires coding skills - I will think about that.

But I like your idea of overriding. I could imagine implementing some of today's "_SEO standards_" as "_example recommendations_" and then allow users to setup the values as they like via some settings panel.

Anyhow, storing settings requires some little research. There are some things to check:
- WHERE to store things (first step will be locally in the browser) 
- HOW to version the data model (to allow future upgrades)

**Comment by bluemoehre** [comment]
It has taken a while, but I am now working on a beta version of this functionality. The first version will be hard-coded in a configuration file, but can easily be modified to create your own local version if needed.

![Over maximum](https://github.com/user-attachments/assets/0ac7629c-c743-4c1d-80ef-1a3d3de8053a)
![Critical range](https://github.com/user-attachments/assets/0c550215-cb81-46c4-8202-83944adda7d0)
![Below minimum](https://github.com/user-attachments/assets/45e3ae38-2bbb-49f6-a08f-cba022a699d8)

Due to limited space & UX, this may not be the final layout - it's just a small preview.
I've been working with it for a few weeks now and I'm trying to keep it at a minimally invasive level while still being able to work with it quickly.

**Comment by bluemoehre** [comment]
I have just released a new version (v1.0.0) that includes the new (beta) features. The extension will be updated automatically as soon as Google has finished testing.

If you (or anyone else) has feedback on specific features, I would be happy if you create new separate tickets.
I hope you will enjoy everything new 🙂
